pkg/server/server.go

```diff
@@ -354,6 +401,17 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 		graph.WithMaxConcurrentReads(s.maxConcurrentReadsForCheck),
 	)
 
+	if s.checkCache != nil {
+		cachedCheckResolver := graph.NewCachedCheckResolver(
+			checkResolver,
+			graph.WithExistingCache(s.checkCache),
+			graph.WithCacheTTL(s.checkQueryCacheTTL),
+		)
+
+		checkResolver.SetDelegate(cachedCheckResolver)
+		defer cachedCheckResolver.Close()
```

Comment:
The cache is shared across requests. You shouldn't close the cache here. The only place where we should be stopping the cache is at server shutdown.

Reply:
+1

Reply:
There is no stopping of the cache here as it is provided with checkCache.

Reply:
See https://github.com/openfga/openfga/blob/c1b773a15e65f2c70ceddc339963326ac53aed21/internal/graph/cached_resolver.go#L139

Reply:
Then why are we calling `defer cachedCheckResolver.Close()` here? It's misleading as a developer reading the code.

Reply:
If we're going to leave this code here like this, without an alternative compositional pattern, then we should at minimum make sure we have a unit test in `cached_resolver_test.go` that asserts that the `CachedCheckResolver.Close()` method doesn't call 'Stop' on the underlying cache. If a developer were to accidentally change that code without realize the impact, then that could be bad and an unintended side-effect.

Reply:
We already have it in https://github.com/openfga/openfga/blob/d9c04f2fd935bb693dbb5fa971dbc3b4050019e9/internal/graph/cached_resolver_test.go#L466 - here we close firstCachedResolver but have secondCachedResolver running with checkCache